core/CMakeLists.txt

```diff
@@ -1047,7 +1047,7 @@ endif ()
 
 if (BUILD_TESTS)
   add_executable(unit_tests unit_tests.c
-    ${CORE_SRCS} ${ARCH_SRCS} ${OS_SRCS})
+    ${CORE_SRCS} ${ARCH_SRCS} ${OS_SRCS} arch/x86_code_test.c)
```

Comment:
It may be worth a comment explaining why just this one test is in a separate file, so someone adding new tests knows which model to follow.

Reply:
Explained in commit message and where we're adding options. I am adding another comment here now.

Reply:
Done.